### PR TITLE
Use endpoints

### PR DESCRIPTION
**Pull request by lmd59**
# Summary
Shifts all front end functionality to use full server operation endpoints rather than local logic with CRUD endpoint interactions.

## New behavior
Front end essentially acts the same with a few changes where spec-based operation functionality differs slightly from the original implemented logic. 
The review functionality no longer asks whether date should be included and is also enabled for active measures.
Server-side $review and $approve operations now validate artifactAssessmentAuthor (reference) input appropriately according to both GET and POST inputs.

## Code changes

Frontend...
- Removes all front end database operation and related files including, `dbOperations.ts`, `mongodb.ts`, updates `.env.example` to remove MONGODB_URI environment variable
- Removes child artifact management (handled by server):`draftHelper.ts`, `serviceUtils.ts`
- Updates `modifyresourceFields.ts` to `versionUtils.ts` to calculate new versions
- Updates all trpc procedures in `draft.ts` and `service.ts` to use server interactions. Note: Future updates could include reorganizing `draft.ts` and `service.ts` to better handle overlapping logic.
- Update to `ReleaseModal.tsx` to remove deletion functionality from release process
- Update `authoring/[resourceType]/[id].tsx` to simplify artifact updates and pass modified values to the `updateDraft` procedure
- Update to `review/[resourceType]/[id].tsx` to change review workflow to use the operation rather than a local update. Enable active artifact reviews and remove date checkbox

Small changes...
- `ResourceInfoCard.tsx` use consolidated trpc procedure
- `pages/[resourceType].tsx` limit getServersideProps fetch to active only
- `authoring/index.tsx` remove uuid setting since the server creates its own id

Server...
- Update `requestSchemas.ts` to validate artifactAssessmentAuthor reference as it would be received from either a GET query parameter or a POST FHIR Parameters object. See https://hl7.org/fhir/R4/search.html#reference for more about how a reference is used in query parameters
- Update Library/Measure services and tests according to new validation logic

# Testing guidance
- `npm run check:all`
- Load your favorite measure bundle(s), including nested child bundle
- `npm run build:all`
- `npm run start:all`
- Test out functionality at http://localhost:3001/mrs/ and make sure it has not changed (other than small ways described above)
- Use insomnia to test $review or $approve GET with artifactAssessmentAuthor input as a string (like a canonical url) and to test POST with artifactAssessmentAuthor as a parameter i.e. within the parameter object: 

> `{ name: 'artifactAssessmentAuthor', valueReference: { reference: 'http://example.org/fhir/Organization/123' } }`

### CODE REVIEW

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Lgtm! All of my comments are just little formatting things, everything looks and works great! HUGE transition!!! 🥳

File: app/src/server/trpc/routers/draft.ts [hunk 1]
```diff
@@ -1,105 +1,209 @@
...
-  updateDraft,
-  deleteDraft,
-  getDraftByUrl,
-  batchDeleteDraft,
-  batchCloneDraft
-} from '../../db/dbOperations';
 import { publicProcedure, router } from '../trpc';
-import { getParentDraftArtifactAndChildren } from '@/util/draftHelper';
-import { modifyResource } from '@/util/modifyResourceFields';
+import { Bundle, OperationOutcome } from 'fhir/r4';
```
[the remark this answers is not in the document]
Comment (lmd59) [reply app/src/server/trpc/routers/draft.ts:4]: Mind if I do the opposite?

[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/server/trpc/routers/draft.ts:4]: Nope! Whatever you want :)

File: app/src/server/trpc/routers/draft.ts [hunk 2]
```diff
@@ -1,105 +1,209 @@
...
       );
 
-      await batchDeleteDraft(draftArtifacts);
+      if (res.status !== 200) {
+        const outcome: OperationOutcome = await res.json();
+        throw new Error(`Received ${res.status} error on $clone:  ${outcome.issue[0].details?.text}`);
+      }
 
-      return { draftId: draftRes.id, children: children };
+      const resBundle: Bundle<FhirArtifact> = await res.json();
```
[the remark this answers is not in the document]
Comment (lmd59) [reply app/src/server/trpc/routers/draft.ts:137]: "

**Review by elsaperelli — APPROVED** [review]
Lgtm!

**Pull request merged**